### PR TITLE
WiP: SciFiDB (formerly ultimate db)

### PR DESCRIPTION
**Pull request by yarikoptic**
Replaces #430, submitted against master
Most likely to fail ATM since WiP

If someone is eager to reduce number of outstanding PRs, I would be ok to have this one merged I guess ... but there is nothing working here yet IIRC

### CODE REVIEW

**Comment by mih** [comment]
Sent a PR against the source branch with a small fix.

**Comment by yarikoptic** [comment]
@hanke thanks! Note that you should be able to push to this branch of mine as well

**Comment by mih** [comment]
Couldn't figure out how. Direct push asked for credentials -- maybe would have just needed to type them in...

**Comment by yarikoptic** [comment]
If using https URL to my clone, then yes, would need to type them. Should work with ssh. See http://git.onerussian.com/?p=etc/base.git;a=blob;f=.gitconfig;h=998a2661aa66a9f5451f6befd592280083e72059;hb=HEAD#l93 for a helper

File: datalad/crawler/nodes/annex.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def __init__(self, path=None,
         self._statusdb = None  # actual DB to be instantiated later
 
         if ultimatedb is None:
-            ultimatedb = cfg.getboolean('crawl', 'annex.ultimatedb', default=False)
+            ultimatedb = cfg.getbool('crawl', 'annex.ultimatedb', default=False)
```
Comment (yarikoptic) [inline datalad/crawler/nodes/annex.py]: forgot to pull ;)
in general -- shouldn't we use `datalad.` prefix in all `cfg.getbool` etc uses for internal config variables?  sample grep showed that it seems to be the case -- ie for internal - `datalad.` prefix. correct @mih?
```
$> git grep 'cfg\.get\S*(' | grep -v -e oldconfig -e test_
datalad/cmd.py:                    cfg.getbool('datalad.log', 'outputs', default=False)
datalad/distribution/create_sibling_github.py:                github_user = cfg.get('hub.oauthtoken', None)
datalad/log.py:        show_timestamps = anything2bool(cfg.get('datalad.log.timestamp', False))
datalad/support/annexrepo.py:                     cfg.getbool("datalad", "repo.direct", default=False)
datalad/support/annexrepo.py:            self._uuid = repocfg.get_value('annex', 'uuid', default='')
```

**Comment by yarikoptic** [comment]
With the current state of metadata support, the usecases for SciFiDB could largely be addressed with extracted metadata (we have now urls, annex.key) alone. So no need to keep dragging this one along IMHO

**Pull request closed without merging**